app/ante/utils_test.go

```diff
@@ -1,104 +1,109 @@
+	suite.app.AccountKeeper.SetParams(suite.ctx, authtypes.DefaultParams())
 	suite.app.EvmKeeper.SetParams(suite.ctx, evmtypes.DefaultParams())
 
-	suite.anteHandler = ante.NewAnteHandler(suite.app.AccountKeeper, suite.app.EvmKeeper, suite.app.SupplyKeeper)
+	suite.encodingConfig = app.MakeEncodingConfig()
+	// We're using TestMsg amino encoding in some tests, so register it here.
+	suite.encodingConfig.Amino.RegisterConcrete(&testdata.TestMsg{}, "testdata.TestMsg", nil)
+
+	suite.anteHandler = ante.NewAnteHandler(suite.app.AccountKeeper, suite.app.BankKeeper, suite.app.EvmKeeper, suite.encodingConfig.TxConfig.SignModeHandler())
 }
 
 func TestAnteTestSuite(t *testing.T) {
 	suite.Run(t, new(AnteTestSuite))
 }
 
-func newTestMsg(addrs ...sdk.AccAddress) *sdk.TestMsg {
-	return sdk.NewTestMsg(addrs...)
+func newTestMsg(addrs ...sdk.AccAddress) *testdata.TestMsg {
+	return testdata.NewTestMsg(addrs...)
 }
 
 func newTestCoins() sdk.Coins {
 	return sdk.NewCoins(ethermint.NewPhotonCoinInt64(500000000))
 }
 
-func newTestStdFee() auth.StdFee {
-	return auth.NewStdFee(220000, sdk.NewCoins(ethermint.NewPhotonCoinInt64(150)))
+func newTestStdFee() legacytx.StdFee {
+	return legacytx.NewStdFee(220000, sdk.NewCoins(ethermint.NewPhotonCoinInt64(150)))
 }
 
 // GenerateAddress generates an Ethereum address.
-func newTestAddrKey() (sdk.AccAddress, tmcrypto.PrivKey) {
+func newTestAddrKey() (sdk.AccAddress, cryptotypes.PrivKey) {
 	privkey, _ := ethsecp256k1.GenerateKey()
 	addr := ethcrypto.PubkeyToAddress(privkey.ToECDSA().PublicKey)
 
 	return sdk.AccAddress(addr.Bytes()), privkey
 }
 
 func newTestSDKTx(
-	ctx sdk.Context, msgs []sdk.Msg, privs []tmcrypto.PrivKey,
-	accNums []uint64, seqs []uint64, fee auth.StdFee,
+	ctx sdk.Context, msgs []sdk.Msg, privs []cryptotypes.PrivKey,
+	accNums []uint64, seqs []uint64, fee legacytx.StdFee,
 ) sdk.Tx {
 
-	sigs := make([]auth.StdSignature, len(privs))
+	sigs := make([]legacytx.StdSignature, len(privs))
 	for i, priv := range privs {
-		signBytes := auth.StdSignBytes(ctx.ChainID(), accNums[i], seqs[i], fee, msgs, "")
+		signBytes := legacytx.StdSignBytes(ctx.ChainID(), accNums[i], seqs[i], 0, fee, msgs, "")
 
 		sig, err := priv.Sign(signBytes)
 		if err != nil {
 			panic(err)
 		}
 
-		sigs[i] = auth.StdSignature{
+		sigs[i] = legacytx.StdSignature{
 			PubKey:    priv.PubKey(),
 			Signature: sig,
 		}
 	}
 
-	return auth.NewStdTx(msgs, fee, sigs, "")
+	return legacytx.NewStdTx(msgs, fee, sigs, "")
 }
 
-func newTestEthTx(ctx sdk.Context, msg evmtypes.MsgEthereumTx, priv tmcrypto.PrivKey) (sdk.Tx, error) {
+func newTestEthTx(ctx sdk.Context, msg *evmtypes.MsgEthereumTx, priv cryptotypes.PrivKey) (sdk.Tx, error) {
 	chainIDEpoch, err := ethermint.ParseChainID(ctx.ChainID())
 	if err != nil {
 		return nil, err
 	}
 
-	privkey, ok := priv.(ethsecp256k1.PrivKey)
-	if !ok {
-		return nil, fmt.Errorf("invalid private key type: %T", priv)
-	}
+	privkey := &ethsecp256k1.PrivKey{Key: priv.Bytes()}
```

Comment:
looks like `ethsecp256k1.PrivKey` is a byte array and not a struct that has `Key` field

Reply:
yes, because the rest of the components are located in other PRs, but it will be a struct